flow/record/tools/rdump.py

```diff
@@ -202,8 +203,11 @@ def main(argv=None):
     selector = make_selector(args.selector, not args.no_compile)
     seen_desc = set()
     count = 0
+    skip = args.skip if args.skip else 0
     with RecordWriter(uri) as record_writer:
         for count, rec in enumerate(record_stream(args.src, selector), start=1):
+            if count <= skip:
+                continue
```

Comment:
Because `skip` is already the count of records you want to skip, you could also opt to just decrease it as a counter:

```suggestion
            if skip > 0:
                skip -= 1
                continue
```

Reply:
This however does break `--count`, as the count is derived from `enumerate()`. A solution would be to get rid of `enumerate()` and track `count` separately when outputting records.

Reply:
`itertools.islice` solves all of these problems i believe.